modules/priceFloors.js

```diff
@@ -0,0 +1,642 @@
+    'endpoint', endpoint => endpoint || {},
+    'enforcement', enforcement => utils.pick(enforcement, [
+      'enforceJS', enforceJS => enforceJS !== false, // defaults to true
+      'enforcePBS', enforcePBS => enforcePBS === true, // defaults to false
+      'floorDeals', floorDeals => floorDeals === true, // defaults to false
+      'bidAdjustment', bidAdjustment => bidAdjustment !== false, // defaults to true
+    ]),
+    'additionalSchemaFields', additionalSchemaFields => typeof additionalSchemaFields === 'object' && Object.keys(additionalSchemaFields).length > 0 ? addFieldOverrides(additionalSchemaFields) : undefined,
+    'data', data => (data && parseFloorData(data, 'setConfig')) || _floorsConfig.data // do not overwrite if passed in data not valid
+  ]);
+
+  // if enabled then do some stuff
+  if (_floorsConfig.enabled) {
+    // handle the floors fetch
+    generateAndHandleFetch(_floorsConfig.endpoint);
+
+    if (!addedFloorsHook) {
+      // register hooks / listening events
+      // when auction finishes remove it's associated floor data
+      events.on(CONSTANTS.EVENTS.AUCTION_END, (args) => delete _floorDataForAuction[args.auctionId]);
+
+      // we want our hooks to run after the currency hooks
+      getGlobal().requestBids.before(requestBidsHook, 50);
+      // if user has debug on then we want to allow the debugging module to run before this, assuming they are testing priceFloors
+      // debugging is currently set at 5 priority
+      getHook('addBidResponse').before(addBidResponseHook, utils.debugTurnedOn() ? 4 : 50);
+      addedFloorsHook = true;
+    }
+  } else {
+    utils.logInfo(`${MODULE_NAME}: Turning off module`);
+
+    _floorsConfig = {};
+    _floorDataForAuction = {};
+
+    getHook('addBidResponse').getHooks({hook: addBidResponseHook}).remove();
+    getGlobal().requestBids.getHooks({hook: requestBidsHook}).remove();
+
+    addedFloorsHook = false;
+  }
+};
+
+/**
+ * @summary Analytics adapters especially need context of what the floors module is doing in order
+ * to best create informed models. This function attaches necessary information to the bidResponse object for processing
+ */
+function addFloorDataToBid(floorData, floorInfo, bid, adjustedCpm) {
+  bid.floorData = {
+    floorValue: floorInfo.matchingFloor,
+    floorRule: floorInfo.matchingRule,
+    floorCurrency: floorData.data.currency,
+    cpmAfterAdjustments: adjustedCpm,
+    enforcements: {...floorData.enforcement},
+    matchedFields: {}
+  };
+  floorData.data.schema.fields.forEach((field, index) => {
+    let matchedValue = floorInfo.matchingData.split(floorData.data.schema.delimiter)[index];
+    bid.floorData.matchedFields[field] = matchedValue;
+  });
+}
+
+/**
+ * @summary simple function which takes the enforcement flags and the bid itself and determines if it should be floored
+ */
+function shouldFloorBid(floorData, floorInfo, bid) {
+  let enforceJS = utils.deepAccess(floorData, 'enforcement.enforceJS') !== false;
```

Comment:
Could the value in `floorData` param be different than the one in `_floorsConfig`?

Reply:
Yeah, 

If the user calls setConfig before the auction and sets it to false,

Then calls another setConfig and sets it to true (or other way around) then it will technically have differing values.

This passes the saved one from the auction offset floor object.